compiler/rustc_ast/src/token.rs

```diff
@@ -191,7 +191,6 @@ pub fn ident_can_begin_expr(name: Symbol, span: Span, is_raw: bool) -> bool {
         || [
             kw::Async,
             kw::Do,
-            kw::Box,
```

Comment:
This is a semantic change for macros, I don't think this should be done right now. I'd need to test it out to check whether it can cause changes to stable-only macros.